marconi-chain-index/test/Spec/Marconi/ChainIndex/Indexers/Utxo/UtxoIndex.hs

```diff
@@ -257,16 +262,46 @@ propUtxoQueryByAddressAndQueryInterval = property $ do
 
   last slotNoFromStorage === endIntervalSlotNo
 
+
+-- Insert Utxo events in storage, and retreive the events by address and slot
+-- Note: The property we are checking is:
+--   - Insert many events with at various chainPoints
+--   - Fetch for all the evenent addresses in the ChainPoint interval [lowChainPoint, highChainPoint]
```

Comment:
evenent -> event

Reply:
Do you mean:
```
data instance StorableQuery UtxoHandle =
   UtxoAddress C.AddressAny | UtxoAddressAtSlot C.AddressAny C.SlotNo
```

I can, the two representations are isomorphic though.
I can compare a query to `UtxoAddress addr Nothing` and another one to `UtxoAddress addr (Just lastSlot)` with the current query.

I think having only one constructor for both cases made the processing easier. Separating them will add an extra resolution step before building the query, or will lead in query duplication. I won't die on this hill though, so if you think it's valuable, I can change it, your take.

Reply:
As we discussed, I borrowed your idea of using the query to generate the where clauses and switched to `NamedQuery`.

Reply:
I'll probably add a few tests tomorrow. I let you decide if we can consider this as resolved or not.